packages/rollout/src/index.ts

```diff
@@ -210,16 +203,23 @@ export const getBinary = async ({
     baseUrlBeta,
     version,
     btcOnly,
+    intermediary = false,
 }: GetBinaryProps) => {
-    // we get info here again, but only as a sanity check.
-    const releaseByFirmware = releases.find(r => versionUtils.isEqual(r.version, version));
+    const parsedFeatures = parseFeatures(features);
     const infoByBootloader = getInfo({ features, releases });
 
-    const parsedFeatures = parseFeatures(features);
-    const parsedReleases = parseReleases(releases);
+    let releaseByFirmware;
 
-    if (!infoByBootloader || !releaseByFirmware) {
-        throw new Error('no firmware found for this device');
+    if (!intermediary) {
+        // we get info here again, but only as a sanity check.
+        releaseByFirmware = releases.find(r => versionUtils.isEqual(r.version, version!));
+
+        if (!infoByBootloader || !releaseByFirmware) {
+            throw new Error('no firmware found for this device');
+        }
+    } else {
+        const fw = await fetchFirmware(`${baseUrl}/firmware/1/trezor-inter-1.8.0.bin`);
+        return { binary: modifyFirmware({ fw, features: parsedFeatures }) };
```

Comment:
looks reasonable, but again already reviewed in https://github.com/trezor/trezor-suite/pull/3215, I am not sure if this PR is a good place for refactoring all the existing code. I need to stop somewhere otherwise this never gets finished  😁